influxdb3_py_api/src/system_py.rs

```diff
@@ -183,6 +187,321 @@ impl PyWriteBatch {
     }
 }
 
+#[pyclass]
+#[derive(Debug)]
+struct PyPluginCallApi {
+    _schema: Arc<DatabaseSchema>,
+    _catalog: Arc<Catalog>,
+    return_state: Arc<Mutex<PluginReturnState>>,
+}
+
+#[derive(Debug, Default)]
+pub struct PluginReturnState {
+    pub log_lines: Vec<LogLine>,
+    pub write_back_lines: Vec<String>,
+    pub write_db_lines: HashMap<String, Vec<String>>,
+}
+
+impl PluginReturnState {
+    pub fn log(&self) -> Vec<String> {
+        self.log_lines.iter().map(|l| l.to_string()).collect()
+    }
+}
+
+pub enum LogLine {
+    Info(String),
+    Warn(String),
+    Error(String),
+}
+
+impl std::fmt::Display for LogLine {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            LogLine::Info(s) => write!(f, "INFO: {}", s),
+            LogLine::Warn(s) => write!(f, "WARN: {}", s),
+            LogLine::Error(s) => write!(f, "ERROR: {}", s),
+        }
+    }
+}
+
+impl std::fmt::Debug for LogLine {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        std::fmt::Display::fmt(self, f)
+    }
+}
+
+#[pymethods]
+impl PyPluginCallApi {
+    fn info(&self, line: &str) -> PyResult<()> {
+        self.return_state
+            .lock()
+            .log_lines
+            .push(LogLine::Info(line.to_string()));
+        Ok(())
+    }
+
+    fn warn(&self, line: &str) -> PyResult<()> {
+        self.return_state
+            .lock()
+            .log_lines
+            .push(LogLine::Warn(line.to_string()));
+        Ok(())
+    }
+
+    fn error(&self, line: &str) -> PyResult<()> {
+        self.return_state
+            .lock()
+            .log_lines
+            .push(LogLine::Error(line.to_string()));
+        Ok(())
+    }
+
+    fn write(&self, line_builder: &Bound<'_, PyAny>) -> PyResult<()> {
+        // Get the built line from the LineBuilder object
+        let line = line_builder.getattr("build")?.call0()?;
+        let line_str = line.extract::<String>()?;
+
+        // Add to write_back_lines
+        self.return_state.lock().write_back_lines.push(line_str);
+
+        Ok(())
+    }
+
+    fn write_to_db(&self, db_name: &str, line_builder: &Bound<'_, PyAny>) -> PyResult<()> {
+        let line = line_builder.getattr("build")?.call0()?;
+        let line_str = line.extract::<String>()?;
+
+        self.return_state
+            .lock()
+            .write_db_lines
+            .entry(db_name.to_string())
+            .or_default()
+            .push(line_str);
+
+        Ok(())
+    }
+}
+
+// constant for the process writes call site string
+const PROCESS_WRITES_CALL_SITE: &str = "process_writes";
+
+const LINE_BUILDER_CODE: &str = r#"
```

Comment:
Is this a stop-gap for now so the line builder doesn't need to be installed?

Reply:
I didn't want to use the existing influxdb3 Python client as it's geared towards making remote requests. I just wanted the simplest possible thing to have some Python lib already in the runtime so that plugin authors could use it.

I think there's going to be some fixed, built-in Python API that we ship with the runtime. For everything else it can be an external library that we can install. Ideally the server would install dependencies for the user automatically when they load the plugin.